webpay/pin/forms.py

```diff
@@ -106,40 +114,61 @@ def clean_pin(self, *args, **kwargs):
             elif res.get('valid'):
                 return pin
 
-        self.add_error_code('WRONG_PIN')
-        raise forms.ValidationError(_('Wrong pin'))
-
-
-class ConfirmPinForm(BasePinForm):
-
-    @sensitive_variables('pin')
-    def clean_pin(self, *args, **kwargs):
-        pin = self.cleaned_data['pin']
-        if self.handle_client_errors(client.confirm_pin(self.uuid, pin)):
-            return pin
-
-        self.add_error_code('PINS_DONT_MATCH')
-        raise forms.ValidationError(_("Pins do not match."))
+        self.add_error_code(msg.WRONG_PIN)
+        raise forms.ValidationError(msg.WRONG_PIN)
 
 
 class ResetPinForm(BasePinForm):
 
+    def __init__(self, *args, **kw):
+        self.session = kw.pop('session')
```

Comment:
I generally don't like it when the session makes it out of the view code. It looks like `user_reset` is the only key that is used. Can that be passed in instead?